pkg/micro/client.go

```diff
@@ -183,6 +181,130 @@ func (c *Client) Request(
 	return &res, nil
 }
 
+// RequestAndSubscribe sends a request and waits for a response on a separate event subject.
+// It returns the raw NATS message, allowing the caller to unmarshal based on their expected type.
+// Default timeouts: 2s for request, 10s for response. Use WithRequestTimeout/WithResponseTimeout to customize.
+//
+// Example:
+//
+//	msg, err := client.RequestAndSubscribe(ctx, address, endpoint, payload, eventSubject,
+//	    micro.WithRequestTimeout(5*time.Second))
+//	if err != nil {
+//	    return err
+//	}
+//	var event iscv1.Event
+//	if err := proto.Unmarshal(msg.Data, &event); err != nil {
+//	    return err
+//	}
+func (c *Client) RequestAndSubscribe(
```

Comment:
Some changes to make this more consistent with `Request`:

```go
func (c *Client) RequestAndSubscribe(
    ctx context.Context,
    requestSubject string,
	  responseSubject string,
	  payload proto.Message,
) (*nats.Msg, error) {}
```

- We'll simplify by using a single timeout for both operations. Callers will have to provide a longer timeout to account for this.
- Extra detail: `responseSubject` doesn't have to have the same address as `requestSubject`.